src/java.desktop/unix/classes/sun/awt/X11/XWindowPeer.java

```diff
@@ -694,12 +694,34 @@ public void checkIfOnNewScreen(Rectangle newBounds) {
                         // Completely on this screen - done!
                         newScreenNum = i;
                         newGC = gds[i].getDefaultConfiguration();
+                        if (target != null
+                                && target.getBackground() != SystemColor.window
+                                && !newGC.isTranslucencyCapable()) {
+                            GraphicsConfiguration[] configs = gds[i].getConfigurations();
+                            for (int j = 0; j < configs.length; ++j) {
+                                if (configs[j].isTranslucencyCapable()) {
+                                    newGC = configs[j];
+                                    break;
+                                }
+                            }
+                        }
```

Comment:
Looks like this code is also repeated on lines 714-724 and can be extracted to a function.